### PR TITLE
S2sport 

### PR DESCRIPTION
**Pull request by jedwards4b**
Changes in cime for s2s model build.  

Test suite: scripts_regression_tests.py
Test baseline: 
Test namelist changes: 
Test status: bit for bit,
Fixes 

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

File: config/ufs/config_files.xml [hunk 1]
```diff
@@ -183,8 +183,8 @@
     <type>char</type>
     <default_value>unset</default_value>
     <values>
-      <value component="cice"      >$SRCROOT/components/cice/</value>
-      <value component="dice"      >$SRCROOT/components/cdeps/dice</value>
+      <value component="cice"      >$SRCROOT/src/model/CICE/</value>
+      <value component="dice"      >$CIMEROOT/src/components/cdeps/dice</value>
```
Comment (jedwards4b) [inline config/ufs/config_files.xml:187]: dice path is wrong

File: scripts/lib/CIME/build.py [hunk 2]
```diff
@@ -139,7 +139,8 @@ def _build_model(build_threaded, exeroot, incroot, complist,
             if "cpl" in l:
                 cpl_in_complist = True
 
-        if cime_model == 'ufs' and not cpl_in_complist:
+#        if cime_model == 'ufs' and not cpl_in_complist:
```
Comment (jedwards4b) [inline scripts/lib/CIME/build.py:142]: Need to fix this.

**Comment by jedwards4b** [comment]
@jgfouca Changes in build.py affect e3sm - the external library paths are now explicit in config_files.xml for each model.

File: config/ufs/config_files.xml [hunk 3]
```diff
@@ -107,7 +107,7 @@
   <entry id="COMP_ROOT_DIR_ATM">
     <type>char</type>
     <values>
-      <value component="datm"      >$SRCROOT/components/cdeps/datm</value>
+      <value component="datm"      >$SRCROOT/src/model/CDEPS/datm</value>
```
Comment (uturuncoglu) [inline config/ufs/config_files.xml:110]: we are checking out CDEPS under components/cdeps for HAFs application. I think this needs to be changed and we need to checkout under src/model/CDEPS/ after this PR. Am I right?

File: config/ufs/config_files.xml [hunk 4]
```diff
@@ -122,8 +122,8 @@
   <entry id="COMP_ROOT_DIR_CPL">
     <type>char</type>
     <values>
+      <value component="cpl" >$SRCROOT/src/model/CMEPS</value>
```
Comment (uturuncoglu) [inline config/ufs/config_files.xml:125]: This is okay but i think still need to have clear distinction between mediator and driver under CIME. So, maybe we could introduce  COMP_ROOT_DIR_DRV etc.

File: config/ufs/config_files.xml (hunk 3 above)
Comment (jedwards4b) [reply config/ufs/config_files.xml:110]: Yes, for s2s I added it as a submodule to ufs-s2s-model, you can do that or keep it in the Externals.cfg file
but you will need the change to manage_externals too.

**Review by uturuncoglu — APPROVED** [review]

File: config/ufs/config_files.xml (hunk 4 above)
Comment (jedwards4b) [reply config/ufs/config_files.xml:125]: I have opened https://github.com/ESMCI/cime/issues/3712 to address this issue.

Comment (uturuncoglu) [reply config/ufs/config_files.xml:125]: Thanks. That is great

**Review by jgfouca — APPROVED** [review]

**Pull request merged**